database/postgres/postgres_test.go

```diff
@@ -307,6 +307,49 @@ func TestPostgres_Lock(t *testing.T) {
 	})
 }
 
+func TestDrop(t *testing.T) {
+	dktesting.ParallelTest(t, specs, func(t *testing.T, c dktest.ContainerInfo) {
+		ip, port, err := c.FirstPort()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		addr := pgConnectionString(ip, port)
+		p := &Postgres{}
+		d, err := p.Open(addr)
+		if err != nil {
+			t.Fatalf("%v", err)
+		}
+		defer d.Close()
+		if err := d.Run(strings.NewReader("CREATE TABLE foo (foo text);")); err != nil {
+			t.Fatalf("expected err to be nil, got %v", err)
+		}
+
+		// drop the table
+		if err := d.Drop(); err != nil {
```

Comment:
Hmm... The migrate is a parent-package which the database package doesn't know, so it wouldn't make sense to use migrate's `Drop()` here. Then the test should be moved to the `migrate_test.go` or another test file in the root package but that doesn't test the specific database implementations.

I guess we would need a suite that does integration-tests on migrate and all the different database implementation which would be a rather significant addition.

I could add such a suite, but it would only cover the `Drop()` method, as I don't have the time to write migrations for each database.

Reply:
Yeah, I wanna avoid scope creep, but also want to prevent a regression. We don't need to test the other methods. e.g. only testing `Drop()` is sufficient. The test in it's current form doesn't cover any different cases than the existing `TestDrop()` in `database/testing/testing.go`.

The quickest way (albeit less clean way) to add the proper regression test would be to import `github.com/golang-migrate/migrate/v4` in `database/testing/testing.go`, create a `migrate.Migrate` instance using `NewWithDatabaseInstance()` in `Test()`, and test `Drop()` using that instance.

I'm all for adding a db driver <-> migrate integration test suite, but I'm not sure about the approach you have for this. The scope for such an integration test suite may also be quite large, so for now, let's add the regression test using the quick and dirty way mentioned above. Then we can add a db driver integration test suite and clean the code in a separate PR.

Thanks again for your work!